### PR TITLE
Dropped Scala 2.10 support

### PR DESCRIPTION
**Pull request by jroper**
* Refactored macros to use quasi quotes
* Fixed a few deprecation warnings

### CODE REVIEW

File: framework/project/plugins.sbt [hunk 1]
```diff
@@ -16,7 +16,7 @@ logLevel := Level.Warn
 
 scalacOptions ++= Seq("-deprecation", "-language:_")
 
-addSbtPlugin("com.typesafe.play" % "interplay" % sys.props.getOrElse("interplay.version", "1.0.2"))
+addSbtPlugin("com.typesafe.play" % "interplay" % sys.props.getOrElse("interplay.version", "1.0.3-SNAPSHOT"))
```
Comment (richdougherty) [inline framework/project/plugins.sbt]: Do you need to publish 1.0.3 before we merge this PR?

**Comment by richdougherty** [comment]
LGTM

File: framework/project/plugins.sbt (hunk 1 above)
Comment (jroper) [inline framework/project/plugins.sbt]: Indeed, I forgot about that.

**Comment by cchantep** [comment]
There is a `2.10.7` version in the replacement

**Comment by dotta** [comment]
Right, I've almost forgot, I'm pushing a PR soon to fix this (James is on vacation for a week, and I wanted to avoid merge conflicts on this one).

**Pull request merged**